src/Microsoft.AspNet.Server.Kestrel/Http/FrameResponseStream.cs

```diff
@@ -110,14 +81,39 @@ public void StopAcceptingWrites()
         {
             // Can't use dispose (or close) as can be disposed too early by user code
             // As exampled in EngineTests.ZeroContentLengthNotSetAutomaticallyForCertainStatusCodes
-            _stopped = true;
+            _state = StreamState.Disposed;
         }
 
         public void Abort()
         {
             // We don't want to throw an ODE until the app func actually completes.
             // If the request is aborted, we throw an IOException instead.
-            _aborted = true;
+            if (_state != StreamState.Disposed)
+            {
+                _state = StreamState.Aborted;
+            }
+        }
+
+        private void ValidateState()
+        {
+            switch (_state)
+            {
+                case StreamState.Open:
+                    return;
+                case StreamState.Disposed:
+                    throw new ObjectDisposedException(nameof(FrameResponseStream));
+                case StreamState.Aborted:
+                    throw new IOException("The request has been aborted.");
+                default:
+                    throw new IOException("Invalid Stream State.");
```

Comment:
Does the compilere require this considering StreamState has only 3 defined values that are already checked?